server/src/HttpServer.hs

```diff
@@ -28,11 +28,14 @@ import qualified Metrics
 new :: Core -> IO Application
 new core =
   scottyApp $ do
-    -- First we check whether the request HTTP method is a recognised HTTP method.
+    -- First middleware logs the time Icepeak takes to handle a request as a
+    -- Prometheus metric.
+    middleware $ measureRequestDuration core
```

Comment:
When doing the split into status and method, we need to move this below the http method sanitization.